drivers/pwm/pwm_nxp_flexio.c

```diff
@@ -130,18 +132,27 @@ static int pwm_nxp_flexio_set_cycles(const struct device *dev,
 
 	pwm_info = &config->pulse_info->pwm_info[channel];
 
-	if ((flags & PWM_POLARITY_INVERTED) == 0) {
-		polarity = FLEXIO_PWM_ACTIVE_HIGH;
-	} else {
+	polarity = (flags & PWM_POLARITY_INVERTED) == 0 ?
+		FLEXIO_PWM_ACTIVE_HIGH : FLEXIO_PWM_ACTIVE_LOW;
+
+	/*
+	 * Checking to see if duty cycle is 0% or 100%
+	 * If so manually keep the GPIO HIGH or LOW.
+	 */
+	if (period_cycles == pulse_cycles) {
 		polarity = FLEXIO_PWM_ACTIVE_LOW;
+		is_only_low_or_high = true;
+	} else if (duty_cycle == period_cycles && !pulse_cycles) {
+		polarity = FLEXIO_PWM_ACTIVE_HIGH;
+		is_only_low_or_high = true;
 	}
```

Comment:
@EmilioCBen the SonarCube is flagging this because it wants an empty "else" here for defensive programming reasons. So end this with:

```
} else {
   /* Some reasonable comment on why there isn't anything in the else block */
}
```

Reply:
Updated the code, just hoping an empty else case doesn't get flagged.